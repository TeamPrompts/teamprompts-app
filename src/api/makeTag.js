import { FITBS, NAME } from './constants';

function makeTag(record) {
  const tag = {
    fitbs: record.get(FITBS),
    id: record.id,
    name: record.get(NAME)
  };
  return tag;
}

export default makeTag;
