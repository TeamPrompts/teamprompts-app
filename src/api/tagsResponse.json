{
  "records": [
    {
      "id": "recmdxpSo7MSWEg00",
      "fields": { "Name": "1:1s", "FITBs": ["recNsahDzKCx0IQPk"] },
      "createdTime": "2019-11-21T23:09:07.000Z"
    },
    {
      "id": "recG0ANQWrp3SRvvY",
      "fields": {
        "Name": "decision making",
        "FITBs": ["recQ2jud83D1dCxr4", "recqeN7TDq4kdhFah"]
      },
      "createdTime": "2019-12-03T22:55:46.000Z"
    },
    {
      "id": "recEZjB6WTnRt6aon",
      "fields": { "Name": "interview" },
      "createdTime": "2019-11-21T23:09:22.000Z"
    },
    {
      "id": "rechbGc1FLmwxb9Hg",
      "fields": {
        "Name": "measurement",
        "FITBs": [
          "reckkQSV8NuHKOAHr",
          "recBKiB7u7DLP6QPe",
          "recrYoglrdF32klOv",
          "recCWoOnmN4aEi5dT",
          "recD5UagMbU0oW464",
          "recglWG0JkW6S4z5T"
        ]
      },
      "createdTime": "2019-11-21T23:09:03.000Z"
    },
    {
      "id": "reciH81wZgz0d5Pt6",
      "fields": { "Name": "misc", "FITBs": ["recvaoKn8ZW4rHGMX"] },
      "createdTime": "2019-12-12T08:07:55.000Z"
    },
    {
      "id": "recBxMIOsQqBhGSaQ",
      "fields": {
        "Name": "product management",
        "FITBs": ["recnGYhq8E9LeRPlW", "recNsahDzKCx0IQPk"]
      },
      "createdTime": "2019-11-21T22:01:53.000Z"
    },
    {
      "id": "recTqD26ElVvXKVad",
      "fields": {
        "Name": "retrospectives",
        "FITBs": [
          "rec99C9lxMl6aSedq",
          "rec590aP1VRIE8f23",
          "recNsahDzKCx0IQPk",
          "recevUpbRFDbVHx7L",
          "reck1NSBKTnmLD6XU"
        ]
      },
      "createdTime": "2019-11-21T22:01:53.000Z"
    },
    {
      "id": "rec1f3sLCcKszXSi4",
      "fields": { "Name": "reviewing decisions" },
      "createdTime": "2019-11-22T22:11:52.000Z"
    },
    {
      "id": "recmdwXtLaj7hls73",
      "fields": {
        "Name": "strategy",
        "FITBs": [
          "receM4Gus7HH43oLM",
          "recBKiB7u7DLP6QPe",
          "recm6xCbOJSJZ6jwY",
          "recCWoOnmN4aEi5dT",
          "recD5UagMbU0oW464",
          "recglWG0JkW6S4z5T"
        ]
      },
      "createdTime": "2019-11-21T22:01:53.000Z"
    }
  ]
}
