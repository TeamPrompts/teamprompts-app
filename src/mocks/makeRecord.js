function makeRecord(element) {
  return {
    get: field => element.fields[field],
    id: element.id
  };
}

export default makeRecord;
