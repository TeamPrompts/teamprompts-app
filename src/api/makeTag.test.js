import makeRecord from '../mocks/makeRecord';
import makeTag from './makeTag';
import { records } from './tagsResponse';

test('tags to match snapshot', () => {
  const tags = records.map(makeRecord).map(makeTag);

  expect(tags).toMatchSnapshot();
});
