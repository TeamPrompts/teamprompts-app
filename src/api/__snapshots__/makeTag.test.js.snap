// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tags to match snapshot 1`] = `
Array [
  Object {
    "fitbs": Array [
      "receM4Gus7HH43oLM",
      "rec99C9lxMl6aSedq",
      "rec590aP1VRIE8f23",
      "recNsahDzKCx0IQPk",
    ],
    "id": "recTqD26ElVvXKVad",
    "name": "retrospectives",
  },
]
`;
